src/sage/rings/polynomial/laurent_polynomial_ring_base.py

```diff
@@ -0,0 +1,535 @@
+# sage.doctest: optional - sage.modules
+r"""
+Ring of Laurent Polynomials (base class)
+
+If `R` is a commutative ring, then the ring of Laurent polynomials in `n`
+variables over `R` is `R[x_1^{\pm 1}, x_2^{\pm 1}, \ldots, x_n^{\pm 1}]`.
+We implement it as a quotient ring
+
+.. MATH::
+
+    R[x_1, y_1, x_2, y_2, \ldots, x_n, y_n] / (x_1 y_1 - 1, x_2 y_2 - 1, \ldots, x_n y_n - 1).
```

Comment:
This is not true: It is not in the category of quotients, which reflects that it doesn't have methods like `ambient()` and `lift()`, nor does it reflect the internal structure. We have implemented it as a localization. Although it is fine to say it is isomorphic to this quotient ring; not that anything needs to be said here.

Reply:
Not my text